api/src/test/java/org/apache/commons/rdf/api/AbstractGraphTest.java

```diff
@@ -228,6 +231,128 @@ public void getTriplesQuery() throws Exception {
         assertEquals(3, graph.getTriples(null, member, null).count());
     }
 
+    @Test
+    public void addBlankNodesFromMultipleGraphs() {
+
+        try {
+            Graph g1 = createGraph1();
+            Graph g2 = createGraph2();
+            Graph g3 = factory.createGraph();
+
+            copyTriples(g1, g3);
+            copyTriples(g2, g3);
+
+            IRI name = factory.createIRI("http://xmlns.com/foaf/0.1/name");
+
+            final Map<String, BlankNodeOrIRI> whoIsWho = new HashMap<>();
+            System.out.println(g3.getTriples(null, name, null).count());
+            // NOTE: sequential as our HashMap is not thread-safe
```

Comment:
OK, agreed to change into ConcurrentHashMap - but only to check the bug is not elsewhere. (done)

This worked, so I suspect the problem was only in the receiving end of storing into `HashMap`. I will add a separate PR to make Graph `Iterable`, which should hopefully reduce the risk of others doing the same mistake for basic operations.

What I would get was inconsistent results from the `HashMap`, e.g. it would say `.size()` as 3, but when you iterate, only get 2 values back, on second run it would work, and third run get a different number, etc. I did also get problems in `copyTriple` without `.sequential`, but that is because the receiving `HashSet` in `GraphImpl.add()` is not thread-safe.

As for `GraphImpl`, I deliberately didn't make it thread-safe or synchronized, as there was no such requirement from the contract (and hence simple should not go beyond the contract, as any thread-safety issues should be flagged for all implementations).